.github/workflows/java-gradle-base.yaml

```diff
@@ -44,33 +44,41 @@ on:
         required: false
         type: string
         default: "."
-
     secrets:
-      signing-key-id:
-        description: "Key id for signing the Sonatype publication."
-        required: true
-      signing-password:
-        description: "Password for signing the Sonatype publication."
+      GOOGLE_WORKLOAD_IDENTITY_PROVIDER:
+        description: The workload identity provider to use for fetching secrets
         required: true
-      signing-secret-key-ring:
-        description: "Key ring (base64 encoded) for signing the Sonatype publication."
+      GOOGLE_SERVICE_ACCOUNT:
+        description: The service account to use to fetch the secrets
         required: true
-      sonar-organization:
-        description: "Organization for Sonarcloud"
+      GOOGLE_PROJECT_ID:
+        description: The id of the project which contains the secrets
         required: true
-      sonar-token:
-        description: "Token for Sonarcloud."
-        required: true
-
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-java-gradle-base
   cancel-in-progress: ${{ github.ref_name != github.event.repository.default_branch }}
-
 jobs:
   build:
     name: Build
     runs-on: ubuntu-22.04
     steps:
+      - name: Authenticate with GCloud
+        uses: google-github-actions/auth@v2
+        id: auth
+        with:
+          workload_identity_provider: ${{ secrets.GOOGLE_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.GOOGLE_SERVICE_ACCOUNT }}
+      - name: Fetch secrets from GSM
+        id: fetch-secrets
+        uses: google-github-actions/get-secretmanager-secrets@v2
+        with:
+          secrets: |-
+            signing-key-id:${{ secrets.GOOGLE_PROJECT_ID }}/SONATYPE_SIGNING_KEY_ID
+            signing-password:${{ secrets.GOOGLE_PROJECT_ID }}/SONATYPE_SIGNING_PASSWORD
+            signing-secret-key-ring:${{ secrets.GOOGLE_PROJECT_ID }}/SONATYPE_SIGNING_SECRET_KEY_RING
+            sonar-organization:${{ secrets.GOOGLE_PROJECT_ID }}/SONARCLOUD_ORGANIZATION
+            sonar-token:${{ secrets.GOOGLE_PROJECT_ID }}/SONATYPE_SIGNING_SECRET_KEY_RING
+          export_to_environment: 'true'
```

Comment:
For the build job we don't actually need any secrets, so we could safe on execution time here